salt/utils/__init__.py

```diff
@@ -1333,7 +1333,11 @@ def fopen(*args, **kwargs):
         if not binary:
             kwargs['encoding'] = 'utf-8'
```

Comment:
This is wrong. `__salt_system_encoding__` should be used instead. Since you're updating this function could you take care of this too @twangboy?